### PR TITLE
Fix Progress Bar in Notebook

### LINKED ISSUE #791: Double printing progress bar
If you force stop a cell in jupyter notebook and then try to rerun, the progress bar get printed on a new line for each update.

Screen Shot 2019-10-29 at 4 37 33 PM

It looks like this can be potentially be fixed by adding `tqdm._instance.clear()` after initializing the progress according to this comment: https://github.com/tqdm/tqdm/issues/375#issuecomment-536659188

here is the text code I ran for the screenshot above

```import featuretools as ft
es = ft.demo.load_mock_customer(return_entityset=True, n_transactions=1000000)
feature_matrix, feature_defs = ft.dfs(entityset=es,
                                       target_entity="customers",
                                       verbose=True)```

### PR DESCRIPTION
**Pull request by jeff-hernandez**
This closes #791 by preventing an interrupted progress bar from double printing when running in a jupyter notebook.

### CODE REVIEW

File: featuretools/computational_backends/calculate_feature_matrix.py [hunk 1]
```diff
@@ -247,6 +247,7 @@ def calculate_feature_matrix(features, entityset=None, cutoff_time=None, instanc
         tqdm_options.update({'file': open(os.devnull, 'w'), 'disable': False})
 
     progress_bar = make_tqdm_iterator(**tqdm_options)
+    progress_bar._instances.clear()
```
Comment (casperdcl) [inline featuretools/computational_backends/calculate_feature_matrix.py:250]: this line should be removed

File: featuretools/computational_backends/calculate_feature_matrix.py [hunk 2]
```diff
@@ -299,7 +300,6 @@ def calculate_feature_matrix(features, entityset=None, cutoff_time=None, instanc
 
     progress_bar.refresh()
```
Comment (casperdcl) [inline featuretools/computational_backends/calculate_feature_matrix.py:301]: this line should be removed

**Comment by rwedge** [comment]
@casperdcl, I implemented those changes on a different branch, tqdm-suggestions, but it did not resolve the issue

Running this code in a notebook:

```python
import featuretools as ft
import tqdm

print(tqdm.__version__)

es = ft.demo.load_mock_customer(return_entityset=True)

cutoff_time = es['transactions'].df[["transaction_id", "transaction_time"]]

ft.dfs(entityset=es, target_entity='transactions', cutoff_time=cutoff_time, verbose=True)
```

Interrupting the kernel during the `dfs` call and rerunning `dfs` caused the progress bar to print on a new line each update

Screen Shot 2020-04-30 at 12 42 56 PM

**Comment by casperdcl** [comment]
Thanks @rwedge - I suspect this may be a `pandas` wrapper issue. Can you try interrupting and re-running:

```tqdm
from tqdm import trange
for i in trange(int(1e9)):
    pass
```

**Comment by rwedge** [comment]
@casperdcl - running the code you provided, interrupting the loop, then re-running the loop does not result in unexpected progress bar behavior.  It looks normal.

**Comment by casperdcl** [comment]
Great so this is a wrapper issue; will open a new issue upstream

**Comment by rwedge** [comment]
Thanks for helping look into this!

**Comment by casperdcl** [comment]
@kmax12 I assume the reason you became [my first ever sponsor](https://github.com/sponsors/casperdcl)? :heart: :confetti_ball:

**Comment by kmax12** [comment]
@casperdcl the reason is that I've been a longtime user of tqdm and wanted to support you. the help here is above and beyond. we really appreciate it. thanks!

**Comment by casperdcl** [comment]
:D I'd like to claim that I have free time I'd like to donate to the world as an act of kindness but the truth is more like this:

xkcd#356

I mean; I wrote the `tqdm.pandas` wrapper years ago, continuously supported since then, and to date still have never used it myself.

**Comment by casperdcl** [comment]
Hmm I don't get it:

```python
#from tqdm.auto import tqdm
from tqdm import tqdm
import pandas as pd
from random import random
import warnings

with warnings.catch_warnings():
    warnings.filterwarnings('ignore', category=FutureWarning)
    tqdm.pandas()

df = pd.DataFrame((random() for _ in range(10)) for _ in range(int(1e6)))

try:
    df.progress_apply(lambda x: x**x ** (1/x) ** x ** (1/x) ** x ** (1/x))
except KeyboardInterrupt:
    print("finished with %d instances left" % len(tqdm._instances))
```
```
60%|██████    | 6/10 [00:02<00:01,  2.77it/s]
finished with 0 instances left
```

----

Also tried (`conda create -n ft featuretools tqdm ipykernel`):

----

```python
#from tqdm.auto import tqdm
from tqdm import tqdm
import featuretools as ft

es = ft.demo.load_mock_customer(return_entityset=True)
cutoff_time = es['transactions'].df[["transaction_id", "transaction_time"]]

try:
    ft.dfs(entityset=es, target_entity='transactions', cutoff_time=cutoff_time, verbose=True)
except KeyboardInterrupt:
    print("finished with %d instances left" % len(tqdm._instances))
```
```
Built 33 features
Elapsed: 00:03 | Progress:   9%|▊         finished with 0 instances left
Elapsed: 00:03 | Progress:   9%|▉         
```

----

screenshot

**Comment by rwedge** [comment]
@casperdcl thanks for looking into this, I've opened a new issue

**Pull request merged**